torch/nn/utils/clip_grad.py

```diff
@@ -34,6 +37,11 @@ def clip_grad_norm_(parameters: _tensor_or_tensors, max_norm: float, norm_type:
         total_norm = max(p.grad.detach().abs().max().to(device) for p in parameters)
     else:
         total_norm = torch.norm(torch.stack([torch.norm(p.grad.detach(), norm_type).to(device) for p in parameters]), norm_type)
+    if error_if_nonfinite and (total_norm.isnan() or total_norm.isinf()):
+        raise ValueError(
+            f'The norm of order {norm_type} for a gradient from `parameters` '
```

Comment:
It would probably be nice if this error message said which of the parameters has a non-finite norm. I'm happy to make that change, but it would impact performance a bit since we would have to check each individual norm, rather than just `total_norm`